game-app/game-headed/src/main/java/games/strategy/triplea/ui/menubar/DebugMenu.java

```diff
@@ -38,43 +28,27 @@ public final class DebugMenu extends JMenu {
 
     setMnemonic(KeyEvent.VK_D);
 
-    frameVisitors.forEach(tripleAFrameConsumer -> tripleAFrameConsumer.accept(frame));
+    DebugMenuInfo.visit(frame);
 
-    debugOptions.stream()
-        .sorted(Comparator.comparing(DebugOption::getName))
-        .forEach(
-            option -> {
-              final JMenu playerDebugMenu = new JMenu(option.getName());
-              add(playerDebugMenu);
-              renderDebugOption(option.getOptions()).forEach(playerDebugMenu::add);
-            });
-    if (frameVisitors.isEmpty() && debugOptions.isEmpty()) {
+    DebugMenuInfo.runForOption(
+        (name, options) -> {
+          final JMenu playerDebugMenu = new JMenu(name);
+          add(playerDebugMenu);
+          renderDebugOption(options).forEach(playerDebugMenu::add);
+        });
+    if (DebugMenuInfo.isEmpty()) {
```

Comment:
I don't love that we're now relying on global state in DebugMenuInfo.

Is there a way to structure this so that the DebugMenuInfo is an object with non-static methods on it that's passed to the constructor of DebugMenu? Then, we wouldn't be introducing and relying on global state that has questionable lifetime.

Reply:
I don't like it either, but this is what the class was already doing before my changes, it just was part of the same class. If you have a better idea I'd be happy to implement it. For now I just kept is as-is and added a separate class so the code compiles

Reply:
I guess my thought is if we're refactoring the code anyway, why propagate the bad pattern?

Looking at how it's called, right now `ProLogUi.registerDebugMenu();` is called from `AbstractProAi()`.

I guess the idea is if there's any ProAI, we get a debug menu. Perhaps this can be done by DebugMenu itself? It already takes a TripleAFrame object, from which we should be able to get access to the game data and see if there's any Pro AI players and call some code from ProLogUI to make the Hard AI menu. I don't know if we even need this "registration" mechanism. We could just create the relevant menus at the time when we create the menu bar, based on relevant conditions ("is there a pro ai")?

Reply:
Good point didn't think about that. I'm worried about merge conflicts though if this stays open for too long so if this is simple to implement we can ship it within this PR otherwise a follow-up PR would also be an option. I believe you know more about the whole environment so it's your call

Reply:
I agree that merge conflicts could be a problem, BUT:
  1. I personally don't have any planned changes right now to UI
  2. We're not currently getting other contributions to UI code either

So given that, I think it would be good to try to give this a shot as part of this PR. It would be a great clean up over the current state and it seems now is a good time to do it given your refactoring is running into this. Do you want to give it a shot? If it ends up being more involved than we're thinking, we could still reconsider.

Reply:
If you can give me a starting point where the `TripleAFrame` stores information about AI I would give it a shot. Might take me a couple of days until I get to it though.

Reply:
I think `TripleAFrame` holds `GameData` from which you can get the `PlayerList` and iterate over `GamePlayers`. Then you can call `getPlayerType()` which should could be used. Alternatively, it also has access to `LocalPlayers` which is a collection of `Player` objects, which you can check for `instanceof AbstractProAi`. I think the latter is probably the better way to do it. Hope that helps!